### PR TITLE
Add test to check device mapping option

### PR DESCRIPTION
**Pull request by harish-24**
Add test to check device mapping option with available namespace

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
```
# avocado run memory/ndctl.py:test_namespace_memmap
Error running method "pre_tests" of plugin "fetchasset": 'Attribute' object has no attribute 'id'
JOB ID     : fa8fb2bdf6d8ebdcfe32ced6e14747727072677f
JOB LOG    : /home/harish/avocado-fvt-wrapper/results/job-2020-02-11T12.39-fa8fb2b/job.log
 (1/1) memory/ndctl.py:NdctlTest.test_namespace_memmap: PASS (0.57 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.67 s
```

File: memory/ndctl.py [hunk 1]
```diff
@@ -405,6 +405,30 @@ def test_namespace_modes(self):
...
             self.fail("Namespace for %s mode failed!" % failed_modes)
 
+    def test_namespace_devmap(self):
+        """
+        Test metadata device mapping option with a namespace
+        """
+        region = self.get_default_region()
+        m_map = self.params.get('map', default='mem')
+        size_align = self.get_size_alignval()
+        namespace_size = self.params.get('size', default=134217728)
```
Comment (narasimhan-v) [inline memory/ndctl.py]: @harish-24 just a suggestion: 
can 134217728 be expanded for better understanding, or can it be documented ?

File: memory/ndctl.py [hunk 2]
```diff
@@ -405,6 +405,30 @@ def test_namespace_modes(self):
...
+        if namespace_size and ((namespace_size % size_align) != 0):
```
Comment (narasimhan-v) [inline memory/ndctl.py]:
```
>>> bool (4 % 2)
False
>>> bool (4 % 3)
True
```

File: memory/ndctl.py [hunk 3]
```diff
@@ -405,6 +405,30 @@ def test_namespace_modes(self):
...
+            self.cancel("Size value not %d aligned %d \n",
```
Comment (narasimhan-v) [inline memory/ndctl.py]: is `Size value %d not %d aligned` better ?

File: memory/ndctl.py (hunk 1 above)
Comment (narasimhan-v) [inline memory/ndctl.py]: better to use it as int too, to be on safer side.

**Comment by harish-24** [comment]
@narasimhan-v Pushed with the changes. Thanks.

**Comment by narasimhan-v** [comment]
@harish-24 please paste job.log with final code, and we are ready for merge.

**Comment by harish-24** [comment]
Job log: [devmap.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4212964/devmap.log)

**Pull request merged**